### PR TITLE
Repeated params must correspond in override

### PR DESCRIPTION
**Pull request by som-snytt**
Refchecks runs after elimRepeated and did not
error on an attempt to override RepeatedParam with Seq.

Also show RepeatedParam in error message for double definition.

Fixes https://github.com/lampepfl/dotty/issues/12662

Resubmits https://github.com/lampepfl/dotty/pull/13248 which was opened against old master.

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/typer/RefChecks.scala [hunk 1]
```diff
@@ -336,6 +336,11 @@ object RefChecks {
         next == other || isInheritedAccessor(next, other)
       }
 
+    def incompatibleRepeatedParam(member: Symbol, other: Symbol): Boolean =
```
Comment (odersky) [inline compiler/src/dotty/tools/dotc/typer/RefChecks.scala:339]: A more efficient version which does not need `nestedZip` would be:
```scala
      def incompatibleRepeatedParam(member: Symbol, other: Symbol): Boolean =
        member.is(Method, butNot = JavaDefined)
        && other.is(Method, butNot = JavaDefined)
        && atPhase(typerPhase):
          !member.info.paramInfoss.corresponds(other.info.paramInfoss): (infos1, infos2) =>
               infos1.nonEmpty && infos1.last.isRepeatedParam
            == infos2.nonEmpty && infos1.last.isRepeatedParam
```

File: compiler/src/dotty/tools/dotc/core/Decorators.scala [hunk 2]
```diff
@@ -234,6 +234,22 @@ object Decorators {
     def nestedExists(p: T => Boolean): Boolean = xss match
       case xs :: xss1 => xs.exists(p) || xss1.nestedExists(p)
       case nil => false
+    def nestedZipExists(yss: List[List[U]])(f: (T, U) => Boolean): Boolean =
+      @tailrec def zipExists(p1: List[T], p2: List[U]): Boolean =
```
Comment (odersky) [inline compiler/src/dotty/tools/dotc/core/Decorators.scala:238]: I think we can do without `nestedZipExists` for the time being. I'd remove it in the interest of keeping the codebase shorter.

File: compiler/src/dotty/tools/dotc/typer/ErrorReporting.scala [hunk 3]
```diff
@@ -125,7 +125,7 @@ object ErrorReporting {
     /** Explain info of symbol `sym` as a member of class `base`.
      *   @param  showLocation  if true also show sym's location.
      */
-    def infoString(sym: Symbol, base: Type, showLocation: Boolean): String =
+    def infoString(sym: Symbol, base: Type, showLocation: Boolean): String = atPhase(Phases.typerPhase) {
```
Comment (odersky) [inline compiler/src/dotty/tools/dotc/typer/ErrorReporting.scala:128]: I am a bit nervous about doing this phase switch for all possible messages. What if symbols get created or changed in an intermediate phase? What does Scala 2 do in this case?

Alternatively, we could also do the `atPhase(typerPhase)` just for the
```
          overrideError("cannot override because erased signatures conflict in repeated parameter")
```

**Review by odersky — CHANGES_REQUESTED** [review]
Otherwise looks good.

File: compiler/src/dotty/tools/dotc/typer/ErrorReporting.scala (hunk 3 above)
Comment (som-snytt) [reply compiler/src/dotty/tools/dotc/typer/ErrorReporting.scala]: It should have been using `DoubleDefinition` which is clever enough. This happens in erasure in Scala 2, where it is slightly more natural. So now, for the `override` case, it `overrides nothing`, otherwise it is `Name clash between defined and inherited member`.

**Comment by som-snytt** [comment]
Thanks again for fruitful review. I spent some time to learn again what the moving parts are, but it still feels awkward to do this narrow check in these two spots, namely for matching methods and double definitions.

**Comment by som-snytt** [comment]
`tests/neg/override-scala2-macro` was the other error message that showed `Seq[A]` instead of `A*`. It's an edge case which is left untouched now.

**Review by odersky — APPROVED** [review]
Sorry for getting to this only now. I did not realize the ball was back in my court here!

Everything LGTM now.
